common/lanelet2_extension/include/lanelet2_extension/regulatory_elements/SignalizedIntersection.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+/*
+ * Copyright (C) 2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include <lanelet2_core/primitives/RegulatoryElement.h>
+#include <boost/algorithm/string.hpp>
+#include <lanelet2_core/utility/Units.h>
+#include <lanelet2_extension/regulatory_elements/CarmaTrafficLight.h>
+#include <unordered_set>
+
+namespace lanelet
+{
+/**
+ * @brief Represents signalized intersection on the road. 
+ *
+ * SignalizedIntersection consists of ENTRY, EXIT, INTERIOR lanelets that are saved as parameters. 
+ * Although this class doesn't manage it, entry lanelets are expected to have CarmaTrafficSignal class to represent the traffic light.
+ * Therefore, this class can be understood as merely recording of which lanelets are part of the intersection, but traffic signal timer
+ * or which entry correlates to which exit information is handled by each CarmaTrafficSignal object itself.
+ *
+ * @ingroup RegulatoryElementPrimitives
+ * @ingroup Primitives
+ */
```

Comment:
This comment should go above the class itself. It reads like it applies to the enum at the moment